### ISSUE TITLE
 Invalid output and model

### ISSUE THREAD

**Issue by kkkmax** [takiyu/portrait_matting#18]

**Comment by YealT** [comment]
> 嗨，我们通过培训获得了模型文件，然后使用模型文件来测试任何照片，但最后我们得到了一个灰色图像，我不知道是什么导致了以下图片：
> ![4](https://user-images.githubusercontent.com/42377340/50784370-6bb13900-12e8-11e9-9af7-4dbaad5e701d.png)
> 
> 我们使用它的方式如下：
> 
> 训练模型命令：
> $ Python scripts / train.py --mode seg --out result-seg
> 
> 训练38纪元后的结果图（max_iteration = 30000）
> ![五](https://user-images.githubusercontent.com/42377340/50784469-ad41e400-12e8-11e9-8fef-16a0f0b6dc03.png)
> 
> ![6](https://user-images.githubusercontent.com/42377340/50807531-b3aa7d00-1335-11e9-9ba3-03a6463ef58e.png)
> ![7](https://user-images.githubusercontent.com/42377340/50807539-ba38f480-1335-11e9-80b6-d2667afd139d.png)
> 
> 在此输出目录中，您可以看到：
> 模型文件“model_best”和训练过程生成文件“snapshot_epoch_x”
> 
> 打开这两个文件，可以看到里面的结构，如下图所示
> 
> ![8](https://user-images.githubusercontent.com/42377340/50807550-bf963f00-1335-11e9-8d04-a6535ce587cf.png)
> 
> 测试模型命令：
> $ python scripts / test.py --mode seg --model_path ./result-seg/model_best --model_mode seg -i my.jpg 
> 结果是看到第一个灰色图
> 
> 尝试再次运行该命令会产生以下错误：
> $ python scripts / test.py --mode seg --model_path ./result-seg/snapshot_epoch_35 --model_mode seg -i M_81_0183.jpg
> 
> ![6](https://user-images.githubusercontent.com/42377340/50807757-7db9c880-1336-11e9-9930-0de9bf0dadac.png)
> 
> 我们检查了输入参数并逐一进行比较。
> 
> 并且读取的模型文件在加载过程中输出每层的W和B值的平方和。可以看出，他们中的大多数都是0。
> ![6](https://user-images.githubusercontent.com/42377340/50807830-cec9bc80-1336-11e9-9e7c-1252a80c0196.png)
> 
> 看到这个结果让我很难过。我再次做了以下尝试：
> 
> 我后来训练了。
> Mode = seg + 
> Mode = seg_tri 
> 训练是正常的，但测试结果和mode = seg 
> 相同的问题：
> 
> 1. 给定任何输入的肖像，{seg，seg +}测试结果图像的颜色是灰色的。
>    {seg_tri}测试结果图像颜色为白色
>    2）{seg，seg +，seg_tri}我从模型中读取的路径为：“。/ result / model_best”
> 2. {seg，seg +，seg_tri}仅保存模型大小约500k
> 3. 我打印了由模型保存的网络的每一层的持久性参数：
>    W（weigths）在所有层中几乎都是0 
>    （偏差）在解码器的最后几层中具有值。
>    这看起来非常不合理
> 
> 我不熟悉这个Chainer，我很难完成整个过程。看到一些指导，我将非常感激和高兴。

@kkkmax  Hello, I have encountered the same problem. Can you solve it? Is it convenient to leave a contact?

**Comment by YealT** [comment]
> I'm not sure if I can't solve another method I'm going to try, you leave your other contact so I can reach you, for example wechat ID??

My wechat is yoni_mmm,think you!

**Comment by YealT** [comment]
> I'm not sure if I can't solve another method I'm going to try, you leave your other contact so I can reach you, for example wechat ID??

My Wechat is yoni_mmm ,thank you

**Comment by billzw** [comment]
> _No description provided._

same question:

Given a portrait of any input, the color of the {seg, seg+} test result image is gray.
{seg_tri}Test result image color is white

Have you solved this problem?

@kkkmax

**Comment by vlordier** [comment]
You need to train the model properly first, using a pre-trained FCN model.
See https://github.com/takiyu/portrait_matting/issues/22

This is the sort of masks (trimap) you can generate after a just few epochs and 2h of training on a p3.8xlarge AWS GPU instance.

![vis_epoch=20_idx=73](https://user-images.githubusercontent.com/5443125/64420125-d5176080-d09e-11e9-8ab3-6cfe42158817.jpg)
